src/material/core/datetime/date-selection.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DateAdapter} from '@angular/material/core';
+import {Subject} from 'rxjs';
+
+export abstract class MatDateSelection<D> {
+  valueChanges = new Subject<void>();
+
+  constructor(protected readonly adapter: DateAdapter<D>) {}
+
+  dispose() {
+    this.valueChanges.complete();
+  }
+
+  abstract add(date: D | null): void;
+  abstract clone(): MatDateSelection<D>;
```

Comment:
I kind of wonder if this method is really necessary. I suspect the thinking was to make a copy of the selection model to show preview states (e.g. hovering over a cell in the calendar, but not actually committing that selection). but I feel like that could just be accomplished with a method like `preview(date: D): S`... however that would make it necessary to parameterize the class on 2 types D (date) and S (selection).

Lets leave this method out and worry about it when we get to a point that we need it.